### PR TITLE
Add Astro Colony

### PR DESCRIPTION
**Pull request by Limmek**
# Description

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [x] You verify that the start command applied does not use a shell script
  * [x] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [x] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [x] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

**Comment by gOOvER** [comment]
this Egg will Not Work, when Port is Set ;)

**Comment by Limmek** [comment]
I have corrected the miss formulating 😅

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    "author": "limmek@swedon.se",
+    "description": "Explore endless Universe and create the most efficient constellation of colonies. Build conveyor systems to automate production. Take care of Astronauts and their needs! Establish new colonies as part of intergalactic simulation.",
+    "features": [
+        "steam_disk_space"
+    ],
+    "docker_images": {
+        "wine": "ghcr.io\/parkervcp\/yolks:wine_latest"
+    },
+    "file_denylist": [],
+    "startup": "wine64 .\/AstroColony\/Binaries\/Win64\/AstroColonyServer.exe -log -SteamServerName=\\\"${SERVER_NAME}\\\" -port=${SERVER_PORT} -QueryPort=${SERVER_QUERY_PORT}",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:18]: use wine here not wine64, as wine now by default is in 64 bit mode

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        "startup": "{\r\n    \"done\": \"server create success\"\r\n}",
+        "logs": "{}",
+        "stop": "^C & ^C"
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:23]: Are you certian this works? else use `^^C`

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
... libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\ncreate_server_settings_ini() {\r\n    cat > ServerSettings.ini <<EOF\r\n[\/Script\/AstroColony.EHServerSubsystem]\r\nServerPassword=${SERVER_PASSWORD}\r\nMapName=${SERVER_MAP}\r\nMaxPlayers=${SERVER_MAXPLAYERS}\r\nSavegameName=${SERVER_GAMESAVE}\r\nShouldLoadLatestSavegame=${SERVER_LOADLATESTSAVEGAME}\r\nAdminList=${SERVER_ADMINS}\r\nSharedTechnologies=${SERVER_SHAREDTECHNOLOGIES}\r\nOxygenConsumption=${SERVER_OXYGENCONSUMPTION}\r\nFreeConstruction=${SERVER_FREECONSTRUCTION}\r\nEOF\r\n}\r\n\r\nFILE=\/mnt\/server\/AstroColony\/Saved\/Config\/WindowsServer\/ServerSettings.ini\r\nif [ -f \"$FILE\" ]; then\r\n    echo \"Config already exists, skipping\"\r\nelse \r\n    echo \"Config does not yet exist, making one\"\r\n    mkdir -p \/mnt\/server\/AstroColony\/Saved\/Config\/WindowsServer\/\r\n    cd \/mnt\/server\/AstroColony\/Saved\/Config\/WindowsServer\/\r\n    create_server_settings_ini\r\nfi\r\n\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:27]: Please leave the full steamcmd instal script there, do not cut some things off, also WINDOWS_INSTALL must be set as an egg variable not here.

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+            "container": "ghcr.io\/pterodactyl\/installers:debian",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:28]: please use ghcr.io/parkervcp/installers:debian, the pterodactyl builds somethimes fail for weeks

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+            "user_editable": true,
+            "rules": "required|string|max:30",
+            "field_type": "text"
+        },
+        {
+            "name": "Server Query Port",
+            "description": "UDP port for Steam server list features.",
+            "env_variable": "SERVER_QUERY_PORT",
+            "default_value": "27015",
+            "user_viewable": true,
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:58]: Users should not be able to change this only see.

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            "user_editable": true,
+            "rules": "required|integer",
+            "field_type": "text"
+        },
+        {
+            "name": "Server Password",
+            "description": "Set the game password. Leave empty to make server public.",
+            "env_variable": "SERVER_PASSWORD",
+            "default_value": "SomePassword",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:67]: Please leave the default empty here

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            "name": "Free Construction",
+            "description": "",
+            "env_variable": "SERVER_FREECONSTRUCTION",
+            "default_value": "0",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|boolean",
+            "field_type": "text"
+        }
+    ]
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:153]: Dont't forget the windows install variable here

Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:153]: Were is the AUTO_UPDATE variabele?

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Max Players",
+            "description": "Amount of max allowed players.",
+            "env_variable": "SERVER_MAXPLAYERS",
+            "default_value": "5",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|integer|max:5",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:90]: Please set a min and max or a between on this

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json (hunk 1 above)
Comment (gOOvER) [inline game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:18]: You missunderstood m; when -port is set and an other port than 7777 is used, you can not connect. remove -port

**Review by gOOvER — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json (hunk 2 above)
Comment (Limmek) [reply game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:23]: Yes it works but i have changed it looks cleaner

File: game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json (hunk 3 above)
Comment (Limmek) [reply game_eggs/steamcmd_servers/astro_colony/egg-astro-colony.json:27]: Makes sense. I was using the icarus egg as reference first but now it is based on wine and steamcmd install script.

**Review by QuintenQVD0 — APPROVED** [review]

**Comment by QuintenQVD0** [comment]
LGTM! and thank you!

**Pull request merged**